java/ql/lib/change-notes/2022-08-09-android-manifest-new-class-and-predicates.md

```diff
@@ -0,0 +1,8 @@
+---
+category: feature
+---
+* Added a new predicate, `requiresPermissions`, in the `AndroidComponentXmlElement` and `AndroidApplicationXmlElement` classes to detect if the element has explicitly set a value for its `android:permission` attribute. Also added `override` annotation to the pre-existing `requiresPermissions` predicate in the `AndroidProviderXmlElement` class.
+* Added a new predicate, `hasAnIntentFilterElement`, in the `AndroidComponentXmlElement` class to detect if a component contains an intent filter element.
+* Added a new predicate, `hasExportedAttribute`, in the `AndroidComponentXmlElement` class to detect if a component has an `android:exported` attribute.
+* Added a new class, `AndroidCategoryXmlElement`, to represent a category element in an Android manifest file. Also added a new predicate, `getCategoryName`, in this class to get the name of the category element.
```

Comment:
```suggestion
* Added a new class, `AndroidCategoryXmlElement`, to represent a category element in an Android manifest file.
```

When adding a class, it's reasonable to expect it'll have a number of predicates, no need to mention them all. If users are interested, once they know the class exists, they can check for themselves.

Reply:
Fixed in commit [cf88d01](https://github.com/github/codeql/pull/9983/commits/cf88d01af67e0e3a4d0f66b699178d39c4846bf6).